File: src/main/java/diff/patch/accessors/SetSetter.java
package diff.patch.accessors;

import diff.patch.Setter;

import java.util.Set;

/**
 *
 * @author sebasjm <smarchano@primary.com.ar>
 */
public class SetSetter<Element> implements Setter<Element> {
    final Set<Element> set;
    final Element element;

    public SetSetter(Set<Element> set, Element element) {
        this.set = set;
        this.element = element;
    }

    @Override
    public void set(Element value) {
        if (element != null) {
            if (value == null) {
                set.remove(element);
            } else {
                set.add(element);
            }
        }
    }
    
}

File: src/main/java/diff/patch/accessors/SetSetter.kt
package diff.patch.accessors

import diff.patch.Setter

/**

 * @author sebasjm <smarchano></smarchano>@primary.com.ar>
 */
class SetSetter<Element: Any>(internal val set: MutableSet<Element>, internal val element: Element?) : Setter<Element> {

    override fun set(value: Element) {
        if (element != null) {
            if (value == null) {
                set.remove(element)
            } else {
                set.add(element)
            }
        }
    }

}
